modules/account-lib/src/coin/baseCoin/ed25519KeyPair.ts

```diff
@@ -74,4 +74,34 @@ export abstract class Ed25519KeyPair implements BaseKeyPair {
 
   /** @inheritdoc */
   abstract getKeys(): any;
+
+  /**
+   * Generates a signature for an arbitrary string with the current private key using the ed25519 public-key signature
+   * system.
+   *
+   * @param {string} message to produce a signature for
+   * @returns {Uint8Array} The signature produced for the message
+   * @throws if there is no private key
+   */
+  signMessage(message: string): Uint8Array {
+    const messageToSign = toUint8Array(Buffer.from(message).toString('hex'));
+    const { prv } = this.keyPair;
+    if (!prv) {
+      throw new Error('Missing private key');
+    }
+    return nacl.sign.detached(messageToSign, nacl.sign.keyPair.fromSeed(toUint8Array(prv)).secretKey);
+  }
+
+  /**
+   * Verifies a message signature using the current public key.
+   *
+   * @param {string} message signed
+   * @param {Uint8Array} signature to verify
+   * @returns {boolean} True if the message was signed with the current key pair
+   */
+  verifySignature(message: string, signature: Uint8Array): boolean {
+    const messageToVerify = toUint8Array(Buffer.from(message).toString('hex'));
```

Comment:
Buffer.from(message) is going to read it as utf-8 text - is that what you intended?

Reply:
```
> a = '1234'
'1234'
> Buffer.from(a).toString('hex')
'31323334'
```